src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -311,14 +360,53 @@ private static Identifiable<?> getIdentifiable(Network network, ContingencyEleme
         return identifiable;
     }
 
+    public void clean(LfNetwork lfNetwork, boolean removeConnectedAtOneSideBranches) {
+        // update branches to open if they are connected to buses in contingency. This is an approximation:
+        // these branches are indeed just open at one side.
+        for (String busId : busIdsToLose) {
+            LfBus bus = lfNetwork.getBusById(busId);
+            if (bus != null) {
+                if (bus.isSlack()) {
+                    // slack bus disabling is not supported
+                    // we keep the slack bus enabled and the connected branches
+                    LOGGER.error("Contingency '{}' leads to the loss of a slack bus: not supported", contingency.getId());
+                    status = Status.NOT_SUPPORTED;
+                } else {
+                    bus.getBranches().forEach(branch -> branchIdsToOpen.add(branch.getId()));
+                }
+            }
+        }
+
+        // we remove disconnected branches or not present in this synchronous component.
+        // if boolean removeConnectedAtOneSideBranches, we remove the branches connected only at one side.
+        Set<String> branchesToRemove = new HashSet<>();
+        for (String branchId : branchIdsToOpen) {
+            LfBranch lfBranch = lfNetwork.getBranchById(branchId);
+            if (lfBranch == null) {
+                branchesToRemove.add(branchId); // disconnected branch
+                continue;
+            }
+            if (!lfBranch.isConnectedAtBothSides() && removeConnectedAtOneSideBranches) {
+                branchesToRemove.add(branchId); // branch connected only on one side
+            }
+        }
+        branchIdsToOpen.removeAll(branchesToRemove);
+    }
+
     public Optional<LfContingency> toLfContingency(LfNetwork network) {
+        // Clean propagated contingency with LfNetwork.
+        clean(network, false);
```

Comment:
toLfContingency purpose is just to convert a PropagatedContingency to a LfContingency and should never mutate the PropagatedContingency. clean method mutates branchIdsToOpen.